modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskServiceImpl.java

```diff
@@ -525,7 +527,7 @@ private TaskDataProvider findProviderByName(DataSource ds) {
             errors.addAll(taskValidator.validateAction(action, channel));
             TriggerEvent trigger = triggerEventService.getTrigger(task.getTrigger());
             Map<Long, TaskDataProvider> providers = getProviders(task);
-            ActionEvent actionEvent = channel.getAction(action);
+            ActionEvent actionEvent = ChannelUtils.getAction(channel, action);
```

Comment:
@ngraczewski I'd say you should rather refactor/create packages as necessary, rather than move the logic to the utils, when there's a circular dependency. @pgesek Can you answer/comment on that?